libs/api/domains/national-registry/src/lib/resolvers/person.resolver.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  CurrentUser,
+  IdsAuthGuard,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import type { User as AuthUser, User } from '@island.is/auth-nest-tools'
+import { Inject, UseGuards } from '@nestjs/common'
+import {
+  Args,
+  Context,
+  Parent,
+  Query,
+  ResolveField,
+  Resolver,
+} from '@nestjs/graphql'
+import { ApiScope } from '@island.is/auth/scopes'
+import { Audit } from '@island.is/nest/audit'
+import {
+  Birthplace,
+  Citizenship,
+  Custodian,
+  Person,
+  PersonBase,
+  Spouse,
+} from '../shared/models'
+import { NationalRegistryService } from '../nationalRegistry.service'
+import type { SharedPerson } from '../shared/types'
+import { Housing } from '../shared/models/housing.model'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+import { Name } from '../shared/models/name.model'
+
+@UseGuards(IdsAuthGuard, IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.meDetails)
+@Resolver(() => Person)
+@Audit({ namespace: '@island.is/api/national-registry' })
+export class PersonResolver {
+  constructor(
+    private service: NationalRegistryService,
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+  ) {}
+
+  @UseGuards(IdsUserGuard, ScopesGuard)
+  @Scopes(ApiScope.meDetails)
+  @Query(() => Person, {
+    nullable: true,
+  })
+  @Audit({ namespace: '@island.is/api/national-registry' })
+  async nationalRegistryPerson(
+    @CurrentUser() user: AuthUser,
+    @Args('api', { nullable: true }) api?: 'v1' | 'v3',
+  ): Promise<Person | null> {
+    const person = await this.service.getPerson(user.nationalId, api ?? 'v1')
+    return person
+  }
+
+  @ResolveField('custodians', () => [Custodian], {
+    nullable: true,
+  })
+  @Audit()
+  async resolveCustodians(
```

Comment:
FYI `async` is redundant when not using `await` and the promise is just returned.

Reply:
yeah, this is mostly leftovers when I remove the debug logging